rasa/core/policies/memoization.py

```diff
@@ -236,6 +236,16 @@ def predict_action_probabilities(
 
         return result
 
+    def format_tracker_states(self, states):
+        # Format tracker states to human readable format on debug log
+        formatted_states = ''
```

Comment:
I think it would be a great idea to append the formatted states into a list and then join the result. Every time you concatenate a string python goes through all of it as an array. Take it into consideration :)